mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/metadata/DeletedMessageMetadataVault.java

```diff
@@ -25,7 +25,7 @@
 import org.reactivestreams.Publisher;
 
 public interface DeletedMessageMetadataVault {
-    Publisher<Void> store(DeletedMessageWithStorageInformation deletedMessage);
+    Publisher<Void> store(User user, DeletedMessageWithStorageInformation deletedMessage);
```

Comment:
What happens when this user is different than the user field of the deletedMessage?

Looks like you are doing an API change to cope with a test logic issue.